src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -228,6 +230,143 @@ object SnowflakeTelemetry {
     )
     SnowflakeTelemetry.send(conn.getTelemetry)
   }
+
+  // Configuration retrieving is optional for for diagnostic purpose,
+  // so it never raises exception.
+  private[snowflake] def getSystemConfigWithoutTaskInfo(): ObjectNode = {
```

Comment:
Why do we have functions with and without task info ? Can't you just always include the task info if it's present ?